solr/src/main/scala/org/apache/pekko/stream/connectors/solr/impl/SolrFlowStage.scala

```diff
@@ -114,8 +113,12 @@ private final class SolrFlowLogic[T, C](
 
       message.routingFieldValue.foreach { routingFieldValue =>
         val routingField = client match {
-          case csc: CloudSolrClient =>
-            Option(csc.getIdField)
+          case csc: CloudSolrClient => {
+            val docCollection = Option(csc.getZkStateReader.getCollection(collection))
+            docCollection.flatMap { dc =>
+              Option(dc.getRouter.getRouteField(dc))
+            }
```

Comment:
```suggestion
          case csc: CloudSolrClient => {
            val docCollection = Some(csc.getZkStateReader.getCollection(collection))
            docCollection.flatMap { dc =>
              Some(dc.getRouter.getRouteField(dc))
            }
```

`Some` and `None` is a group of subclasses of `Option`, in Scala, we recommend using `Some` rather than directly use `Option`

Reply:
Agreed, unless you are expecting values like `csc.getZkStateReader.getCollection(collection)` to be `null`

Reply:
The whole point is the worry that this Java code could return null. The use of `Option` is deliberate here.